MantidQt/CustomInterfaces/test/ReflCommandsTest.h

```diff
@@ -0,0 +1,321 @@
+#ifndef MANTID_CUSTOMINTERFACES_REFLCOMMANDSTEST_H
+#define MANTID_CUSTOMINTERFACES_REFLCOMMANDSTEST_H
+
+#include <cxxtest/TestSuite.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "MantidAPI/FrameworkManager.h"
+#include "MantidQtCustomInterfaces/Reflectometry/IReflTablePresenter.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflAppendRowCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflClearSelectedCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflCopySelectedCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflCutSelectedCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflDeleteRowCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflExpandCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflExportTableCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflGroupRowsCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflImportTableCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflNewTableCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflOpenTableCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflOptionsCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflPasteSelectedCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflPlotGroupCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflPlotRowCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflPrependRowCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflProcessCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflSaveTableAsCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflSaveTableCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflTableViewPresenter.h"
+
+using namespace MantidQt::CustomInterfaces;
+using namespace Mantid::API;
+using namespace testing;
+
+class MockIReflTablePresenter : public IReflTablePresenter {
+
+public:
+  MOCK_METHOD1(notify, void(IReflTablePresenter::Flag));
+
+private:
+  // Calls we don't care about
+  const std::map<std::string, QVariant> &options() const { return m_options; };
```

Comment:
I tried this out and it works. Here's a mini example.

```cpp
class ReturnValue{
  public:
    // Non-trivial return value. Cannot be default constructed, so gmock gets stuck.
    ReturnValue(int v) : var(v) {}
    int var = 0;
};

class TypeWithReturn {
  public:
    virtual ReturnValue compute() const = 0;
    virtual ~TypeWithReturn(){}
};

class MockTypeWithReturn : public TypeWithReturn {
public:
  MockTypeWithReturn(){
      // This is the critical bit. We express what we want to return in the absence of any other information. This will be the same for all instances of the Mock.
      ON_CALL(*this, compute()).WillByDefault(Return(ReturnValue(1)));
  }
  MOCK_CONST_METHOD0(compute, ReturnValue());
  ~MockTypeWithReturn(){}
};

TEST(demo, try_out_mock){
    // Create the mock as ususal
    NiceMock<MockTypeWithReturn> mock;
    // The important part is the EXPECT_CALL always overrides the WillByDefault from ON_CALL
    EXPECT_CALL(mock, compute()).WillOnce(Return(ReturnValue(0)));
    auto var = mock.compute();
    EXPECT_EQ(var.var, 0);


    EXPECT_TRUE(Mock::VerifyAndClearExpectations(&mock)) << "Mock not used properly";

}
```